fbw-a380x/src/systems/systems-host/systems/Communications/Transponder.ts

```diff
@@ -68,6 +68,20 @@ export class Transponder implements Instrument {
     Wait.awaitSubscribable(GameStateProvider.get(), (v) => v === GameState.ingame).then(() => {
       // set initial squawk code as the cfg file param doesn't seem to work
       SimVar.SetSimVarValue('K:XPNDR_SET', 'number', 0x2000);
+      switch (getFltState()) {
+        case FltState.Hanger:
+        case FltState.Apron:
+          break;
+        case FltState.Taxi:
+        case FltState.Runway:
+        case FltState.Climb:
+        case FltState.Cruise:
+        case FltState.Approach:
+        case FltState.Final:
+        default:
+          this.bus.getPublisher<MfdSurvEvents>().pub('mfd_xpdr_set_auto', true, true);
+          break;
```

Comment:
You can just use the cold and dark start Lvar for this like all the other systems :P.

Reply:
That's not as fun! but I'll refactor